RecoTracker/MkFit/interface/MkFitHitWrapper.h

```diff
@@ -0,0 +1,60 @@
+#ifndef RecoTracker_MkFit_MkFitHitWrapper_h
+#define RecoTracker_MkFit_MkFitHitWrapper_h
+
+#include "DataFormats/Provenance/interface/ProductID.h"
+
+#include <memory>
+#include <vector>
+
+namespace mkfit {
+  class EventOfHits;
+  class TrackerInfo;
+  class Hit;
+  using HitVec = std::vector<Hit>;
+}  // namespace mkfit
+
+class MkFitHitWrapper {
+public:
+  MkFitHitWrapper();
+  MkFitHitWrapper(mkfit::TrackerInfo const& trackerInfo);
+  ~MkFitHitWrapper();
+
+  MkFitHitWrapper(MkFitHitWrapper const&) = delete;
+  MkFitHitWrapper& operator=(MkFitHitWrapper const&) = delete;
+  MkFitHitWrapper(MkFitHitWrapper&&);
+  MkFitHitWrapper& operator=(MkFitHitWrapper&&);
+
+  mkfit::EventOfHits& eventOfHits() { return *eventOfHits_; }
+  mkfit::EventOfHits const& eventOfHits() const { return *eventOfHits_; }
+
+  void setPixelClustersID(edm::ProductID id) { pixelClustersID_ = id; }
+  edm::ProductID pixelClustersID() const { return pixelClustersID_; }
+
+  void setOuterClustersID(edm::ProductID id) { outerClustersID_ = id; }
+  edm::ProductID outerClustersID() const { return outerClustersID_; }
+
+  mkfit::HitVec& pixelHits() { return *pixelHits_; }
+  mkfit::HitVec const& pixelHits() const { return *pixelHits_; }
+
+  mkfit::HitVec& outerHits() { return *outerHits_; }
+  mkfit::HitVec const& outerHits() const { return *outerHits_; }
+
+  std::vector<float>& stripClusterCharge() { return stripClusterCharge_; }
+  void stripClusterChargeCut(float minThreshold, std::vector<bool>& mask) const;
+
+private:
+  std::unique_ptr<mkfit::EventOfHits> eventOfHits_;
+
+  // using unique_ptr to guarantee the address of the HitVec doesn't change in moves
+  // EvenfOfHits relies on that
+  // Vectors are indexed by the cluster index
+  std::unique_ptr<mkfit::HitVec> pixelHits_;
+  std::unique_ptr<mkfit::HitVec> outerHits_;
+
+  std::vector<float> stripClusterCharge_;
```

Comment:
here you make use of `strip` ... sorry for the low level comment, but I think it would be better to have a common and homogenous naming scheme / convention

Reply:
In general I agree on consistency, but this information is specific for phase0/1 strip clusters (do phase2 outer tracker clusters provide the charge information? I did not see in 
https://github.com/cms-sw/cmssw/blob/master/DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h)

I could think of
* be phase1 specific in naming for now, and deal with generality towards phase2 later (the setup is phase1 specific anyway), i.e. rename `outer` to `strip`
* try to be general already now, probably means moving the `stripClusterCharge` into its own Event data product

Reply:
> do phase2 outer tracker clusters provide the charge information? 

The OT readout is binary, so each element is just "on" or "off" and cluster charge would be ~trivially proportional to the cluster size, (single digis carry the overthreshold bit information)

Reply:
Actually (reminding myself), the pixel and strip/outer hits need to be separated at the Event product level anyway for (eventual) initialStepPreSplitting to avoid converting strip/outer hits twice. That would naturally solve the naming issues here.

Reply:
Class has been split for separate event products for pixel and strip, with the `vector<float>`specific for strip also as a separate event data product.